Remotion/Web/Development.WebTesting.UnitTests/Accessibility/AccessibilityResultViolationExtensionTest.cs

```diff
@@ -0,0 +1,225 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Remotion.Web.Development.WebTesting.Accessibility;
+
+namespace Remotion.Web.Development.WebTesting.UnitTests.Accessibility
+{
+  [TestFixture]
+  public class AccessibilityResultViolationExtensionTest
+  {
+    private readonly AccessibilityRuleResult _defaultViolation = new AccessibilityRuleResult (
+        new AccessibilityRule (
+            AccessibilityRuleID.AudioCaption,
+            "Description",
+            AccessibilityTestImpact.Moderate,
+            new[] { AccessibilityTestSuccessCriteria.Wcag_1_3_6 }),
+        new[] { new AccessibilityTestTarget ("xPath", "CSS") },
+        "Html",
+        new[] { new AccessibilityRequirement (AccessibilityCheckID.MetaViewport, "Message", AccessibilityTestImpact.Moderate) },
+        new AccessibilityRequirement[0],
+        new AccessibilityRequirement[0]);
+
+    [Test]
+    public void FilterByCssSelectorAndImpact ()
+    {
+      var filter = new AccessibilityResultFilter
+                   {
+                       IgnoreCssSelectors = { "test" },
+                       IncludeImpact = AccessibilityTestImpact.Moderate
+                   };
+      var violation = CreateViolation (impact: AccessibilityTestImpact.Moderate, targets: new[] { CreateTargetWithXPathAndCss ("test") });
+      var accessibilityResult = CreateAccessibilityResult (violation, _defaultViolation);
+
+      var filteredViolations = accessibilityResult.Violations.Filter (filter);
```

Comment:
Do we actually need a accessibilityresult or should we just have a collection of the correct type?

Reply:
What do you mean exactly? Do you want to

- [ ] Get rid of the CreateAccessibilityResult factory method in these tests?
- [ ] Remove the type `AccessibilityResult` overall?
- [ ] None of the above

Reply:
1. Rename CreateAccessiblityResult to CreateViolations / alternatively, just create an array with the relevant entries if this remains legible. 
2. change var accessiblityResult to hold the collection of Violation objects.
3. Apply Filter() to this variable.
4. Remove no longer needed code from Test, do not change production code.